File: benchmarks/str_concat.lua
-- Двойник str_concat.bsl. Именно `s = s .. piece`, а не table.concat:
-- строки Lua тоже неизменяемы, и сравнивать надо один и тот же способ
-- сборки. (Идиоматичный Lua писал бы через table.concat — он линейный, но
-- тогда это был бы бенчмарк другого алгоритма.)

local piece = "абвгдеёжзийклмнопрстуфхцчшщъыьэюя0123456789"

local started = os.clock()
local text = ""
for _ = 1, 3000 do
    text = text .. piece
end
local elapsed_ms = (os.clock() - started) * 1000

print(string.format("длина (байт): %d", #text))
print(string.format("%.3f", elapsed_ms))

File: benchmarks/str_concat.py
"""Двойник `str_concat.bsl` с последовательной конкатенацией строк."""

from time import perf_counter


PIECE = "абвгдеёжзийклмнопрстуфхцчшщъыьэюя0123456789"


started = perf_counter()
text = ""
for _ in range(10):
    text = ""
    for _ in range(3_000):
        text = text + PIECE
elapsed_ms = (perf_counter() - started) * 1_000

print(f"длина: {len(text)}")
print(f"{elapsed_ms:.3f}")
